### PR TITLE
Fix for Issue 1416

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This is a fix for #1416. Calculation of the histogram was done in parallel without locking.

### CODE REVIEW

File: src/ImageSharp/Processing/Processors/Normalization/GlobalHistogramEqualizationProcessor{TPixel}.cs [hunk 1]
```diff
@@ -123,7 +123,8 @@ public void Invoke(int y)
                     // TODO: We should bulk convert here.
                     var vector = Unsafe.Add(ref pixelBase, x).ToVector4();
                     int luminance = ImageMaths.GetBT709Luminance(ref vector, levels);
-                    Unsafe.Add(ref histogramBase, luminance)++;
+                    ref int histogramAtLuminance = ref MemoryMarshal.GetReference(this.histogramBuffer.Slice(luminance));
+                    Interlocked.Increment(ref histogramAtLuminance);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Processing/Processors/Normalization/GlobalHistogramEqualizationProcessor{TPixel}.cs:127]: Couldn't this have been

```c#
Interlocked.Increment(ref Unsafe.Add(ref histogramBase, luminance))
```

Comment (brianpopow) [reply src/ImageSharp/Processing/Processors/Normalization/GlobalHistogramEqualizationProcessor{TPixel}.cs]: yes right, dont know why i did it so over complicated

Comment (brianpopow) [reply src/ImageSharp/Processing/Processors/Normalization/GlobalHistogramEqualizationProcessor{TPixel}.cs]: not sure why net472 tests are failing now. Its not the histogram tests:

`Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at ImageMagick.PixelCollection.NativeMethods.X86.PixelCollection_ToByteArray(IntPtr Instance, UIntPtr x, UIntPtr y, UIntPtr width, UIntPtr height, IntPtr mapping, IntPtr& exception)
   at ImageMagick.PixelCollection.NativePixelCollection.ToByteArray(Int32 x, Int32 y, Int32 width, Int32 height, String mapping)
  ...`

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Normalization/GlobalHistogramEqualizationProcessor{TPixel}.cs]: It's an intermittent issue with ImageMagick. Maybe caused by OOM but I'm not sure. I just reset when I see it.

**Review by JimBobSquarePants — APPROVED** [review]

**Pull request merged**